src/is-where.js

```diff
@@ -1,28 +1,72 @@
 const entries = require('object.entries');
 
+const ATTRIBUTE_PRESENT = {exists: true};
+
 const _isWhere = (where, target) => {
+  // Check each key from where
   for (const [key, value] of entries(where)) {
+
+    // If the key is set, but value is undefined, we ignore it
+    if (typeof value === 'undefined') {
+      continue;
+    }
+
+    // Allow a way to check for the pressenece of an attribute with that name.
+    if (value === ATTRIBUTE_PRESENT) {
+      if (!(key in target)) {
+        return false;
+      }
+      continue;
+    }
+
+    // Array checks
+    if (Array.isArray(value)) {
```

Comment:
Does this have a test?

Reply:
no it didn't and I'm not sure that we even use this branch for anything...

It seems to be implemented as an "or", was that the intention, should I just remove it?

Reply:
It's supposed to be an AND. https://github.com/mzgoddard/preact-render-spy/commit/4b3ca7607e57aa5b7781d7cedd00344be014ce34#diff-428d6b5dd241183ba23e9ab5fac3deaeR20

Became an OR at a point in its history. Looks like it was during the special casing class commit. https://github.com/mzgoddard/preact-render-spy/commit/cf3cb7401d09657b1d5073c02613369d651e501c#diff-428d6b5dd241183ba23e9ab5fac3deaeR43

It changed from testing if a key had all of the strings contained in it to equaling at least one of the strings.

It originally was there for finding elements that had all of a set of classes.

Reply:
I think we can get rid of it. Couldn't find a test that uses either functionality.

As well you can do AND functionality with multiple chained `.find`s. And you could or by `concat`ing multiple non-chained `.find`s.